### PR TITLE
Extend vocab padding for logits MatMul for fp16 GPT2 GreedySearch

### PR DESCRIPTION
**Pull request by hariharans29**
Extend https://github.com/microsoft/onnxruntime/pull/13686 to GreedySearch

### CODE REVIEW

File: onnxruntime/test/contrib_ops/greedy_search_test.cc [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+  if (HasCudaEnvironment(min_cuda_architecture)) {
+    Ort::SessionOptions session_options;
+#ifdef USE_CUDA
+    Ort::ThrowOnError(OrtSessionOptionsAppendExecutionProvider_CUDA(session_options, 0));
+#endif
+
+    // The following model was obtained by padding the vocabulary size in testdata/transformers/tiny_gpt2_beamsearch_fp16.onnx
+    // (by making beam_size == 1) from 1000 to 1600 (just for illustrative and testing purposes) to see if the beam search
+    // implementation can handle such a scenario
+    Ort::Session session(*ort_env, ORT_TSTR("testdata/transformers/tiny_gpt2_greedysearch_fp16_padded_vocab.onnx"), session_options);
```
[the remark this answers is not in the document]
Comment (hariharans29) [reply onnxruntime/test/contrib_ops/greedy_search_test.cc:63]: Yes. I did not find any tests for GreedySearch. So, added one test. Let me know if I am mistaken.

[the remark this answers is not in the document]
Comment (tianleiwu) [reply onnxruntime/test/contrib_ops/greedy_search_test.cc:63]: I think ye means the unit test failure
```
[ RUN      ] GreedySearchTest.GptGreedySearchFp16_VocabPadded
testdata/transformers/tiny_gpt2_greedysearch_fp16_padded_vocab.onnx failed. File doesn't exist" thrown in the test body.
```

[the remark this answers is not in the document]
Comment (hariharans29) [reply onnxruntime/test/contrib_ops/greedy_search_test.cc:63]: Oh right. Yeah, I pushed it just now.

**Review by tianleiwu — APPROVED** [review]

**Review by wangyems — APPROVED** [review]

**Review by yufenglee — APPROVED** [review]
